### PR TITLE
man: add explanation about target's DefaultDependencies=

### LINKED ISSUE #7113: pid1: specifying DefaultDependencies=no in service unit causes default After= option lost in target unit
### Submission type

 - Bug report

### systemd version the issue has been seen with

v235

### Used distribution

Arch Linux

### In case of bug report: Expected behaviour you didn't see

```
# systemctl list-dependencies --after some.target 
some.target
● └─some.service
```

### In case of bug report: Unexpected behaviour you saw

```
# systemctl list-dependencies --after some.target 
some.target
```

### In case of bug report: Steps to reproduce the problem

Create 2 units: `some.service` and `some.target`:

```
# cat some.service
[Unit]
DefaultDependencies=no

[Service]
ExecStart=/bin/sleep 5566

# cat some.target
[Unit]
Wants=some.service
```

According to `systemd.target(5)`, the default `After=` dependency won't be applied if `DefaultDependencies=no` is specified in the target unit. However, in this case, `DefaultDependencies=no` is specified in the service unit. There is no description about this behavior in `systemd.service(5)`.

### PR DESCRIPTION
**Pull request by yuwata**
Closes #7113.

### CODE REVIEW

File: man/systemd.target.xml [hunk 1]
```diff
@@ -95,7 +95,8 @@
       <listitem><para>Target units will automatically complement all
       configured dependencies of type <varname>Wants=</varname> or
       <varname>Requires=</varname> with dependencies of type
-      <varname>After=</varname>. Note that <varname>Wants=</varname> or
+      <varname>After=</varname> unless <varname>DefaultDependencies=no</varname>
+      is not set in the specified units. Note that <varname>Wants=</varname> or
```
Comment (arvidjaar) [inline man/systemd.target.xml]: I think proper English is "unless is set". Here you effectively have double negation, so the effect - `After` is not added   if `DefaultDependecies=no` is **not** present.

File: man/systemd.unit.xml [hunk 2]
```diff
@@ -297,7 +297,8 @@
     <para>For example, target units will complement all configured
     dependencies of type <varname>Wants=</varname> or
     <varname>Requires=</varname> with dependencies of type
-    <varname>After=</varname>. See
+    <varname>After=</varname> unless <varname>DefaultDependencies=no</varname>
+    is not set in the specified units. See
```
Comment (arvidjaar) [inline man/systemd.unit.xml]: Same comment

File: man/systemd.target.xml (hunk 1 above)
Comment (yuwata) [reply man/systemd.target.xml]: Ahh... Thanks. I will update.

Comment (yuwata) [reply man/systemd.target.xml]: Updated.

**Pull request merged**